lldb/source/Commands/CommandObjectDWIMPrint.cpp

```diff
@@ -117,7 +149,7 @@ bool CommandObjectDWIMPrint::DoExecute(StringRef command,
                                         flags, expr);
       }
 
-      valobj_sp->Dump(result.GetOutputStream(), dump_options);
+      write_valobj_to_result(*valobj_sp, dump_options);
```

Comment:
Suggestion, decompose this. Separately call `Dump`, as it is now, and then following that, the helper function can inspect the `result`'s output by calling `GetOutputData`.

Reply:
That's how I initially implemented it, the problem is that either the hint would be under the output, or we'd need some duplicated setup code on every call site :(